src/lib/ppx_coda/versioned.ml

```diff
@@ -363,24 +366,37 @@ let generate_constructor_decl_decls type_name ctor_decl =
       in
       arg_lets @ result_lets
 
-let generate_contained_type_decls type_decl =
+let generate_constraint_type_decls type_name cstrs =
+  let gen_for_constraint (ty1, ty2, _loc) =
+    List.concat_map [ty1; ty2] ~f:(generate_core_type_version_decls type_name)
+  in
+  List.concat_map cstrs ~f:gen_for_constraint
+
+let generate_contained_type_version_decls type_decl =
   let type_name = type_decl.ptype_name.txt in
-  match type_decl.ptype_kind with
-  | Ptype_abstract ->
-      if Option.is_none type_decl.ptype_manifest then
+  let constraint_type_version_decls =
+    generate_constraint_type_decls type_decl.ptype_name.txt
+      type_decl.ptype_cstrs
+  in
+  let main_type_version_decls =
+    match type_decl.ptype_kind with
+    | Ptype_abstract ->
+        if Option.is_none type_decl.ptype_manifest then
```

Comment:
Nit: This is essentially a match, might be clearer unless we're really concerned about indent nesting.

Reply:
Done.